src/scenes/gameuiscreen.ts

```diff
@@ -1,27 +1,43 @@
 import { ScreenHeight, ScreenWidth } from "core/main";
 import Phaser from "phaser";
 import { assert } from "utilities/utils";
+import { PlaceTile } from "../definitions/placeTile";
 import { int, Nullable, UUID } from "../definitions/utils";
 import { Tile } from "../entities/Tile";
 import TileManager from "../managers/TileManager";
 
 export class GameUiScreen extends Phaser.Scene {
     private uiBackground: Nullable<Phaser.GameObjects.Rectangle>;
     private drawnTilesContainer: Nullable<Phaser.GameObjects.Container>;
+    private dragObj: Nullable<Phaser.GameObjects.Image>;
+    private originalLocations: { [key: string]: [int, int] };
+    private currentTile: PlaceTile;
+    private topLeftX: int;
+    private topLeftY: int;
+    private isDraging: boolean;
+    private static tilePixels: int = 128;
 
     public constructor() {
         super("GameUiScreen");
         this.uiBackground = null;
         this.drawnTilesContainer = null;
+        this.dragObj = null;
+        this.originalLocations = {};
+        this.currentTile = new PlaceTile();
+        this.topLeftX = 0;
+        this.topLeftY = 0;
+        this.isDraging = false;
     }
 
     public init(): void {
-        const tileUpdateListener: UUID = TileManager.onSync.on(() => {
+       /* const tileUpdateListener: UUID = TileManager.onSync.on(() => {
```

Comment:
makes little sense to constanly update the tiles in hand because then if you drag it away it get's updated during the drag and appears again in the hand until you place it.

Reply:
wasn't sure it makes sense that's why i didn't delete it yet

Reply:
My intension was that when client receives tile update from ping, display placed tiles with updated view. If i understand correctly, there shouldnt be a conflict because tile information in tileManager doesnt change if u click it or drag it?

Reply:
Why do you need to update the tiles in hand with a ping? the server doesn't know anything about tiles in hand. I think i don't get the question can you please elaborate. And this wasn't implemented by you it was done by croxy.

Reply:
Im sorry im talking about displaying placed tiles, not tiles in hand.
I didnt see that this is in gameuiscreen, my bad